README.md

```diff
@@ -121,6 +121,45 @@ This function searches the node hash of an application for all nodes that have a
 collect_component_nodes($nodes, Wordpress_app::Web)
 ```
 
+#### `create_component_app`
+
+This function is similiar to the normal application declaration syntax but
+instead of mapping nodes to component instances it maps components to nodes.
+This can make it a lot easier to user applications defined with
+`collect_component_titles` where the same components may be assigned to many
+nodes for availability or scale.  The following two site blocks will result in
+equivalent application declarations:
+
+```puppet
+site {
+  create_component_app('wordpress::app', 'my_wordpress',
+    {'db_password' => 'secret',
+     'components' => {
+       'Wordpress_app::Database' => ['node1.example.com'],
```

Comment:
This *only* works correctly if the resource types are in quotes. Otherwise, the components come out with names like `Wordpress_app::Database[][Wordpress_app-b2eq0e8dan8u9cq.delivery.puppetlabs.net]`, because the component type is `Wordpress_app::Database[]`.